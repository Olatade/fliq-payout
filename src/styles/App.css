.App{
  @apply
  min-h-screen
}

.nav {
  @apply 
  flex
  gap-4
  items-center
  px-4 
  py-4 
  bg-white

  md:px-40
}

.nav__logo{
  @apply h-8
}


.body{
  @apply
  grid
  h-screen
  bg-gray-50
  py-20
  px-4
  sm:px-0
  ;
}

.form{
  @apply container mx-auto max-w-lg bg-white self-start
  rounded-md
}