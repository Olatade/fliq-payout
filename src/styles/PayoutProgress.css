.progress-bar{
  @apply hidden md:block w-full mx-auto
}

.left-pad, .right-pad{ @apply flex-1 }

.step{ @apply flex-1}

.step__inner{
  @apply w-2 h-2 bg-gray-300 mx-auto rounded-full text-lg text-gray-500 flex items-center
}

/* change background when step is active */
.step__inner.active{
  @apply bg-color-primary
}

.progress{
  @apply w-1/6 items-center align-middle content-center flex
}

.progress__inner{
  @apply w-full bg-gray-200 rounded items-center align-middle  flex-1
}

.progress__length{
  @apply bg-color-primary text-xs leading-none py-1 text-center text-gray-700 rounded w-full
}

.step-names{
  @apply flex text-xs content-center text-center pt-1
}
.step-names div{
  @apply w-1/4
}