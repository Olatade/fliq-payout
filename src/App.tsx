import React from 'react';
import logo from './assets/img/fliqpay-logo.png';
import './styles/App.css';
import FormPayoutAmount from './components/form-payout-amount';
import FormPayoutReceipt from './components/form-payout-receipt';
import PayoutReview from './components/payout-review';
import PayoutProgress from './components/Payout-progress';


function App() {
  return (
    <div className="App">

      <header className="nav">
        <img src={logo} className="nav__logo" alt="logo" />
        <PayoutProgress/>
      </header>

      <div className="body">
        <FormPayoutAmount/>
      </div>

    </div>
  );
}

export default App;
