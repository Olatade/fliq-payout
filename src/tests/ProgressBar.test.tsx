import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import App from '../App';
import PayoutProgress from '../components/Payout-progress';
import renderer from 'react-test-renderer';

// make sure every test is starting from thesame starting point
afterEach(() =>{
  cleanup();
})

test('should render payout progress bar', () => {
  const stage = 1
  render(<PayoutProgress stage={stage} />);
  const appElement = screen.getByTestId('payout-progress');
  expect(appElement).toBeInTheDocument();
});

test('matches stage 1 snapshot', () => {
  const stage = 1
  const tree = renderer.create(<PayoutProgress stage={stage} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('matches stage 2 snapshot', () => {
  const stage = 2
  const tree = renderer.create(<PayoutProgress stage={stage} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('matches stage 3 snapshot', () => {
  const stage = 3
  const tree = renderer.create(<PayoutProgress stage={stage} />).toJSON();
  expect(tree).toMatchSnapshot();
});





// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

// test('renders Navigation component', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/Review/i);
//   expect(linkElement).toBeInTheDocument();
// });
