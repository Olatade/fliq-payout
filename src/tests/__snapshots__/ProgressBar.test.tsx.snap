// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches stage 1 snapshot 1`] = `
<div
  className="progress-bar"
  data-testid="payout-progress"
>
  <div
    className="flex"
  >
    <div
      className="left-pad"
    />
    <div
      className="step step-1"
    >
      <div
        className="step__inner active"
      />
    </div>
    <div
      className="progress"
    >
      <div
        className="progress__inner"
      >
        <div
          className="progress__length"
          style={
            Object {
              "width": "0",
            }
          }
        />
      </div>
    </div>
    <div
      className="step step-2"
    >
      <div
        className="step__inner"
      />
    </div>
    <div
      className="progress"
    >
      <div
        className="progress__inner"
      >
        <div
          className="progress__length"
          style={
            Object {
              "width": "0",
            }
          }
        />
      </div>
    </div>
    <div
      className="step step-3"
    >
      <div
        className="step__inner"
      />
    </div>
    <div
      className="progress"
    >
      <div
        className="progress__inner"
      >
        <div
          className="progress__length"
          style={
            Object {
              "width": "0",
            }
          }
        />
      </div>
    </div>
    <div
      className="step step-4"
    >
      <div
        className="step__inner"
      />
    </div>
    <div
      className="right-pad"
    />
  </div>
  <div
    className="step-names"
  >
    <div>
      Amount
    </div>
    <div>
      Receipt
    </div>
    <div>
      Review
    </div>
    <div>
      Pay
    </div>
  </div>
</div>
`;

exports[`matches stage 2 snapshot 1`] = `
<div
  className="progress-bar"
  data-testid="payout-progress"
>
  <div
    className="flex"
  >
    <div
      className="left-pad"
    />
    <div
      className="step step-1"
    >
      <div
        className="step__inner active"
      />
    </div>
    <div
      className="progress"
    >
      <div
        className="progress__inner"
      >
        <div
          className="progress__length"
          style={
            Object {
              "width": "100%",
            }
          }
        />
      </div>
    </div>
    <div
      className="step step-2"
    >
      <div
        className="step__inner active"
      />
    </div>
    <div
      className="progress"
    >
      <div
        className="progress__inner"
      >
        <div
          className="progress__length"
          style={
            Object {
              "width": "0",
            }
          }
        />
      </div>
    </div>
    <div
      className="step step-3"
    >
      <div
        className="step__inner"
      />
    </div>
    <div
      className="progress"
    >
      <div
        className="progress__inner"
      >
        <div
          className="progress__length"
          style={
            Object {
              "width": "0",
            }
          }
        />
      </div>
    </div>
    <div
      className="step step-4"
    >
      <div
        className="step__inner"
      />
    </div>
    <div
      className="right-pad"
    />
  </div>
  <div
    className="step-names"
  >
    <div>
      Amount
    </div>
    <div>
      Receipt
    </div>
    <div>
      Review
    </div>
    <div>
      Pay
    </div>
  </div>
</div>
`;

exports[`matches stage 3 snapshot 1`] = `
<div
  className="progress-bar"
  data-testid="payout-progress"
>
  <div
    className="flex"
  >
    <div
      className="left-pad"
    />
    <div
      className="step step-1"
    >
      <div
        className="step__inner active"
      />
    </div>
    <div
      className="progress"
    >
      <div
        className="progress__inner"
      >
        <div
          className="progress__length"
          style={
            Object {
              "width": "100%",
            }
          }
        />
      </div>
    </div>
    <div
      className="step step-2"
    >
      <div
        className="step__inner active"
      />
    </div>
    <div
      className="progress"
    >
      <div
        className="progress__inner"
      >
        <div
          className="progress__length"
          style={
            Object {
              "width": "100%",
            }
          }
        />
      </div>
    </div>
    <div
      className="step step-3"
    >
      <div
        className="step__inner active"
      />
    </div>
    <div
      className="progress"
    >
      <div
        className="progress__inner"
      >
        <div
          className="progress__length"
          style={
            Object {
              "width": "0",
            }
          }
        />
      </div>
    </div>
    <div
      className="step step-4"
    >
      <div
        className="step__inner"
      />
    </div>
    <div
      className="right-pad"
    />
  </div>
  <div
    className="step-names"
  >
    <div>
      Amount
    </div>
    <div>
      Receipt
    </div>
    <div>
      Review
    </div>
    <div>
      Pay
    </div>
  </div>
</div>
`;
